osquery/tables/system/linux/disk_encryption.cpp

```diff
@@ -110,14 +115,27 @@ QueryData genFDEStatus(QueryContext& context) {
   }
 
   std::map<std::string, Row> encrypted_rows;
-  auto block_devices = SQL::selectAllFrom("block_devices");
-  for (const auto& row : block_devices) {
-    const auto name = (row.count("name") > 0) ? row.at("name") : "";
-    const auto uuid = (row.count("uuid") > 0) ? row.at("uuid") : "";
-    const auto parent_name = (row.count("parent") > 0 ? row.at("parent") : "");
-    genFDEStatusForBlockDevice(
-        name, uuid, parent_name, encrypted_rows, results);
+
+  auto constraint = context.constraints["name"].getAll(EQUALS);
+  auto names = std::vector(constraint.begin(), constraint.end());
```

Comment:
This shouldn't be needed, `constraint` here is a `std::set<std::string>` which can be looped by a ranged for as you're already doing with the vector.

Reply:
Sorry that I did not write a working example in my comment. The comment smjert references is the syntactically correct version of my suggestion. :P